### DISCUSSION TITLE
How can I define a Header() field as required, all openapi headers show as optional

### DISCUSSION THREAD

**Discussion by austenwho** [fastapi/fastapi#4853] [Questions] [answered]
All header fields are interpreted as optional, probably due to the required `default` value in the Header initialization, even when I declare the field parameter in the request definition without using `Optional[...]` I cannot force the field to be required, nor does the openapi generated documentation show the field as required.

```python
@app.get("/")
def required_header(q: str, authentication: str = Header(""), x_api_key: str = Header("")):
    ...
```

Omitting Header's `default` parameter is not optional.

I'm defining a lot of endpoints that have required header parameters (authentication, etc) that I would like:
a) enforced
b) documented
which would be super convenient to define in a Depends function.

How can I make this happen?

**Comment by wu-clan** [comment, marked as answer]
Can this help you? https://fastapi.tiangolo.com/zh/tutorial/dependencies/dependencies-in-path-operation-decorators/#_5

**Reply by austenwho** [reply]
Yes, that solved the problem, thank you!
